File: 8.Best-Practice-e-trucchi-avanzati/objc/CM_HF/CM_HF WatchKit Extension/InterfaceController.m
//
//  InterfaceController.m
//  CM_HF WatchKit Extension
//
//  Created by WhiteTiger "sgama la rete" on 28/02/16.
//  Copyright © 2016 WhiteTiger "sgama la rete". All rights reserved.
//

#import "InterfaceController.h"


@interface InterfaceController()

@end


@implementation InterfaceController

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];
}

- (void)willActivate {
    [super willActivate];
}

- (void)didDeactivate {
    [super didDeactivate];
}

#pragma mark - IBAction Methods

- (IBAction)actionHaptickClick:(id)sender {
    [[WKInterfaceDevice currentDevice] playHaptic:(WKHapticTypeClick)];
}

- (IBAction)actionHaptickSuccess:(id)sender {
    [[WKInterfaceDevice currentDevice] playHaptic:(WKHapticTypeSuccess)];
}

- (IBAction)actionHaptickStop:(id)sender {
    [[WKInterfaceDevice currentDevice] playHaptic:(WKHapticTypeStop)];
}

- (IBAction)actionHaptickRetry:(id)sender {
    [[WKInterfaceDevice currentDevice] playHaptic:(WKHapticTypeRetry)];
}

- (IBAction)actionHaptickNotification:(id)sender {
    [[WKInterfaceDevice currentDevice] playHaptic:(WKHapticTypeNotification)];
}

- (IBAction)actionHaptickFailure:(id)sender {
    [[WKInterfaceDevice currentDevice] playHaptic:(WKHapticTypeFailure)];
}

- (IBAction)actionHaptickStart:(id)sender {
    [[WKInterfaceDevice currentDevice] playHaptic:(WKHapticTypeStart)];
}

- (IBAction)actionHaptickDirectionUP:(id)sender {
    [[WKInterfaceDevice currentDevice] playHaptic:(WKHapticTypeDirectionUp)];
}

- (IBAction)actionHaptickDirectionDown:(id)sender {
    [[WKInterfaceDevice currentDevice] playHaptic:(WKHapticTypeDirectionDown)];
}

@end

File: 8.Best-Practice-e-trucchi-avanzati/swift/CM_HF/CM_HF WatchKit Extension/InterfaceController.swift
//
//  Copyright © 2015 WhiteTiger "sgama la rete". All rights reserved.
//

import WatchKit
import Foundation

class InterfaceController: WKInterfaceController {

    override func awakeWithContext(context: AnyObject?) {
        super.awakeWithContext(context)
    }

    override func willActivate() {
        super.willActivate()
    }

    override func didDeactivate() {
        super.didDeactivate()
    }
    
    // MARK: - IBAction Methods

    @IBAction func actionHaptickClick() {
        WKInterfaceDevice.currentDevice().playHaptic(.Click)
    }
    
    @IBAction func actionHaptickSuccess() {
        WKInterfaceDevice.currentDevice().playHaptic(.Success)
    }
    
    @IBAction func actionHaptickStop() {
        WKInterfaceDevice.currentDevice().playHaptic(.Stop)
    }
    
    @IBAction func actionHaptickRetry() {
        WKInterfaceDevice.currentDevice().playHaptic(.Retry)
    }
    
    @IBAction func actionHaptickNotification() {
        WKInterfaceDevice.currentDevice().playHaptic(.Notification)
    }
    
    @IBAction func actionHaptickFailure() {
        WKInterfaceDevice.currentDevice().playHaptic(.Failure)
    }
    
    @IBAction func actionHaptickStart() {
        WKInterfaceDevice.currentDevice().playHaptic(.Start)
    }
    
    @IBAction func actionHaptickDirectionUP() {
        WKInterfaceDevice.currentDevice().playHaptic(.DirectionUp)
    }
    
    @IBAction func actionHaptickDirectionDown() {
        WKInterfaceDevice.currentDevice().playHaptic(.DirectionDown)
    }
}
